src/pages/Questionnaire.tsx

```diff
@@ -1,20 +1,275 @@
+import { QuestionnaireCard } from "../components/styled/QuestionnaireCard";
+import { Ques } from "../components/Ques";
+import { RoutePaths } from "../common/Routes";
+
+// styled-component styles for Questionnaire Page
+
+const QuesContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: flex-start;
+  align-items: center;
+  height: 100%;
+  width: 100%;
+  padding: 20px;
+  overflow-y: auto;
+`;
+
+const ProgressBar = styled.div`
+  width: 100%;
+  display: flex;
+  justify-content: space-between;
+  margin-bottom: 20px;
+
+  .progress-segment {
+    width: 100%;
+    height: 8px;
+    background-color: #ccc;
+    border-radius: 4px;
+    margin-right: 4px;
+
+    &.filled {
+      background-color: #07889b;
+    }
+  }
+`;
+
+const ProgressIndicator = styled.div`
+  font-size: 1.15rem;
+  color: #07889b;
+  font-weight: bold;
+  margin-bottom: 20px;
+  text-align: center;
+`;
+
+const NavigationButton = styled.button`
+  background-color: #07889b;
+  color: white;
+  border: none;
+  padding: 10px 20px;
+  border-radius: 20px;
+  cursor: pointer;
+  font-size: 1rem;
+  margin: 10px;
+  width: 150px;
+
+  &:hover {
+    background-color: #066a83;
+  }
+
+  &:disabled {
+    background-color: #ccc;
+    cursor: not-allowed;
+  }
+
+  @media (max-width: 768px) {
+    padding: 8px 16px;
+    width: 100px;
+    font-size: 0.9rem;
+  }
+`;
+
+const SubmitButton = styled(NavigationButton).attrs({ as: "a" })`
+  display: inline-block;
+  text-align: center;
+  text-decoration: none;
+`;
+
+const TitleText = styled.h1`
+  color: #07889b;
+  margin-top: 20px;
+  margin-bottom: 20px;
+  text-align: center;
+  font-size: 2.5rem;
+
+  @media (max-width: 768px) {
+    font-size: 2rem;
+    margin-top: 10px;
+    margin-bottom: 10px;
+  }
+`;
+
+const CardText = styled.h5`
+  color: #222831;
+  margin-bottom: 20px;
+  margin-right: 40px;
+  margin-left: 40px;
+  text-align: center;
+  font-size: 1.5rem;
+
+  @media (max-width: 768px) {
+    font-size: 1.25rem;
+    margin-right: 20px;
+    margin-left: 20px;
+  }
+`;
+
+const StartButton = styled(NavigationButton)`
+  width: 25%;
+  text-align: center;
+
+  @media (max-width: 768px) {
+    width: 40%;
+  }
+
+  &:hover {
+    background-color: #f5f5f5;
+    color: #07889b;
+    font-weight: bold;
+    border: solid;
+    border-color: #07889b;
+  }
+`;
+
+// types
+
+enum Type {
+  RADIO = "radio",
+  CHECKBOX = "checkbox",
+  RANGE = "range",
+  DROPDOWN = "dropdown",
+}
+
+interface Option {
+  optionId: number;
+  optionLabel: string;
+}
+
+interface QuesProps {
+  id: number;
+  title: string;
+  type: Type;
+  options: Option[];
+}
 
 export function Questionnaire() {
-  const activeLink = links.find(
-    (link) => window.location.pathname === link.href
+  const { questionId } = useParams<{ questionId: string }>();
+  const navigate = useNavigate();
+  const [currentQuestion, setCurrentQuestion] = useState<number>(
+    questionId ? parseInt(questionId) - 1 : -1
```

Comment:
I am not sure I understand this logic. If the URL has `questionId`, then why are you doing: `parseInt(questionId) - 1` and not just `parseInt(questionId)` ?

Reply:
It was because the questionID from the URL started with an index of 1, but the questions array started at index 0. I've updated the JSON so that the question ID will now start at 0 so it's all the same. The default is still -1 if there is no questionID so that it renders the start-question page first.